basic/src/test/java/org/jboss/hal/testsuite/test/configuration/infinispan/ParametersFactory.java

```diff
@@ -0,0 +1,77 @@
+package org.jboss.hal.testsuite.test.configuration.infinispan;
+
+import org.apache.commons.lang.RandomStringUtils;
+import org.jboss.hal.testsuite.test.configuration.infinispan.cache.container.CacheContainerContext;
+import org.jboss.hal.testsuite.test.configuration.infinispan.cache.container.CustomCacheContainerContext;
+import org.jboss.hal.testsuite.test.configuration.infinispan.cache.container.DefaultCacheContainer;
+import org.jboss.hal.testsuite.test.configuration.infinispan.cache.type.DistributedCacheTypeContext;
+import org.jboss.hal.testsuite.test.configuration.infinispan.cache.type.InvalidationCacheTypeContext;
+import org.jboss.hal.testsuite.test.configuration.infinispan.cache.type.LocalCacheTypeContext;
+import org.jboss.hal.testsuite.test.configuration.infinispan.cache.type.ReplicatedCacheTypeContext;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+public class ParametersFactory {
+
+    private OnlineManagementClient client;
+
+    private ParametersFactory(OnlineManagementClient client) {
+        this.client = client;
+    }
+
+    private static ParametersFactory INSTANCE = null;
+
+    public static ParametersFactory newInstance(OnlineManagementClient client) {
+        if (INSTANCE == null) {
+            return new ParametersFactory(client);
+        }
+        return INSTANCE;
+    }
+
+    public Collection cacheContainerTable() {
+        CacheContainerContext hibernate = DefaultCacheContainer.HIBERNATE;
+        CacheContainerContext web = DefaultCacheContainer.WEB;
+        CacheContainerContext ejb = DefaultCacheContainer.EJB;
+        CacheContainerContext server = DefaultCacheContainer.SERVER;
+        CacheContainerContext custom = new CustomCacheContainerContext(client, "custom_" + RandomStringUtils.randomAlphanumeric(7));
+        return Arrays.asList(new Object[][]{
+                {hibernate},
+                {web},
+                {ejb},
+                {server},
+                {custom}
+        });
+    }
+
+    public Collection containerTypeMatrix() {
```

Comment:
Please, add JavaDoc.

Reply:
Please add JavaDoc describing the purpose of this method as well as the structure of the provided collection.